coord.py

```diff
@@ -1,31 +1,422 @@
+        	+ x8frac * x8poly * np.power (x, 8.0)
+        	
+        #/* Calculate longitude */
+        philambda[1] = lambda0 + x1frac * x\
+        	+ x3frac * x3poly * np.power (x, 3.0)\
+        	+ x5frac * x5poly * np.power (x, 5.0)\
+        	+ x7frac * x7poly * np.power (x, 7.0)
+        	
+        return philambda
+
+    
+
+    """/*
+    * LatLonToUTMXY
+    *
+    * Converts a latitude/longitude pair to x and y coordinates in the
+    * Universal Transverse Mercator projection.
+    *
+    * Inputs:
+    *   lat - Latitude of the point, in radians.
+    *   lon - Longitude of the point, in radians.
+    *   zone - UTM zone to be used for calculating values for x and y.
+    *          If zone is less than 1 or greater than 60, the routine
+    *          will determine the appropriate zone from the value of lon.
+    *
+    * Outputs:
+    *   xy - A 2-element array where the UTM x and y values will be stored.
+    *
+    * Returns:
+    *   The UTM zone used for calculating the values of x and y.
+    *
+    */"""
+    def LatLonToUTMXY (self, lat, lon, zone):
+        xy = self.MapLatLonToXY(lat, lon, self.UTMCentralMeridian(zone))
+
+        #/* Adjust easting and northing for UTM system // WITH ED's weird numbers. */
+        xy[0] = xy[0] * self.UTMScaleFactor + 499600
+        xy[1] = xy[1] * self.UTMScaleFactor + -1800
+        if xy[1] < 0.0:
+            xy[1] = xy[1] + 10000000.0
+
+        return xy, zone
+    
+    
+    """/*
+    * UTMXYToLatLon
+    *
+    * Converts x and y coordinates in the Universal Transverse Mercator
+    * projection to a latitude/longitude pair.
+    *
+    * Inputs:
+    *	x - The easting of the point, in meters.
+    *	y - The northing of the point, in meters.
+    *	zone - The UTM zone in which the point lies.
+    *	southhemi - True if the point is in the southern hemisphere;
+    *               false otherwise.
+    *
+    * Outputs:
+    *	latlon - A 2-element array containing the latitude and
+    *            longitude of the point, in radians.
+    *
+    * Returns:
+    *	The function does not return a value.
+    *
+    */"""
+    def UTMXYToLatLon (self, x, y, zone, southhemi):
+        x -= 499600
+        x /= self.UTMScaleFactor
+        	
+        #/* If in southern hemisphere, adjust y accordingly. */
+        if southhemi:
+            y -= 10000000.0
+
+        y -= -1800	
+        y /= self.UTMScaleFactor 
+        
+        cmeridian = self.UTMCentralMeridian (zone)
+        latlon = self.MapXYToLatLon (x, y, cmeridian)
+        	
+        return latlon
+
+    
+
+
+
+
+    def lat_lon_to_xz(self, lat, lon):
+        zone = 36 #caucasus hardcoded zone 36?? save for later np.floor ((lon + 180.0) / 6) + 1
+        utmxy, zone = self.LatLonToUTMXY(np.deg2rad(lat), np.deg2rad(lon), zone)
+
+        return utmxy[1]-4998115, utmxy[0]-599517
+
+    def xz_to_lat_lon(self, x, z):
+        zone = 36
+        southhemi = False
+
+        utmy = x + 4998115
```

Comment:
What do these magic numbers do? A comment or a name for the constant would be helpful for folks maintaining this later.